x/feegrant/types/codec.go

```diff
@@ -7,19 +7,22 @@ import (
 
 // RegisterCodec registers the account types and interface
 func RegisterCodec(cdc *codec.Codec) {
+	cdc.RegisterInterface((*isFeeAllowance_Sum)(nil), nil)
 	cdc.RegisterInterface((*exported.FeeAllowance)(nil), nil)
+	cdc.RegisterConcrete(&FeeAllowance_BasicFeeAllowance{}, "cosmos-sdk/ProtoBasicFeeAllowance", nil)
 	cdc.RegisterConcrete(&BasicFeeAllowance{}, "cosmos-sdk/BasicFeeAllowance", nil)
 	cdc.RegisterConcrete(&PeriodicFeeAllowance{}, "cosmos-sdk/PeriodicFeeAllowance", nil)
-
 	cdc.RegisterConcrete(FeeGrantTx{}, "cosmos-sdk/FeeGrantTx", nil)
 }
 
```

Comment:
I would imagine we would need a custom `Codec` interface here that knows how to encode/decode all interface types. See `x/gov`, `x/auth`, or `x/evidence` for guidance. This ties into my previous comments on not using `x.Marshal`.